### ISSUE TITLE
Remove CLM4.0

### ISSUE THREAD

**Issue by billsacks** [ESCOMP/ctsm#455]
We'd like to remove support for CLM4.0 soon, largely in order to clean up build-namelist.

We are awaiting final approval from Stan Soloman, Dan Marsh and Hanli Liu that they won't need CLM4 for WACCM-X for any releases after CESM2.1.

**Comment by billsacks** [comment]
@ekluzek I'm starting to work on this (finally!). My initial sense is that we can remove much/all of buildcpp and configure (though I'll give these a closer look to see if anything needs to be maintained). 

Once CLM4.0 is removed, `CLM_CONFIG_OPTS` will just be used for `-phys`. I propose introducing a new env_run.xml variable for this important option. I propose the name `CTSM_PHYSICS_VERSION` (though I'm not at all tied to that name, and am open to suggestions). Currently this xml variable could have values "clm4_5" or "clm5_0"; soon it could also have something like "nwp0_1". It looks like moving the `-phys` option out of `CLM_CONFIG_OPTS` into its own variable will only have minor implications for the code – basically, it will change (simplify) how it is obtained in buildnml and related scripts.

@ekluzek let me know if you have any thoughts on this.

**Comment by ekluzek** [comment]
I think this makes sense. My initial thought was that the "-phys" option would move over to CLM_BLDNML_OPTS. But, I think you are right it should be it's own thing. In either case it would only affect "build-namelist", so could change at run-time.

I do wonder about starting to introduce env variables that start with CTSM. The current ones all start with CLM, so you can do "./xmlquery --partial CLM" and get all of them. Now, you'll need to check both CLM and CTSM. Perhaps we should change the others to CTSM as well? Although as a transition maybe introducing the first one makes sense.

The rest of the name makes sense to me. We should just start to think about how to handle the transition of CLM_ to CTSM_...

**Comment by billsacks** [comment]
Thanks, @ekluzek . I don't want to tackle renaming all of the xml variables in this tag, so if you think consistency is important, I can go with `CLM_PHYSICS_VERSION` instead of `CTSM_PHYSICS_VERSION` for now.

**Comment by billsacks** [comment]
For now I'm using `CLM_PHYSICS_VERSION`.

**Comment by ekluzek** [comment]
Yes, I think that's best. Then we can convert all of them to the CTSM_* at the same time.

**Comment by billsacks** [comment]
@ekluzek FYI: For namelist definition / defaults files and the (hopefully-soon-to-be-deleted) config definition file, I'm using a suffix of ctsm rather than clm. I think you raise a good point about consistency for things that a user is likely to interact with, but for these, which very few people deal with, it seemed better to go ahead and make this change, rather than having to go back through the renaming exercise yet again later (since these file names appear in quite a few places).

**Comment by ekluzek** [comment]
That makes perfect sense to me.
